projects/observability/src/pages/explorer/explorer-service.ts

```diff
@@ -37,14 +47,20 @@ export class ExplorerService {
     );
 
     return forkJoinSafeEmpty(filterStrings$).pipe(
-      map(filterStrings => ({
-        navType: NavigationParamsType.InApp,
-        path: '/explorer',
-        queryParams: {
+      map(filterStrings => {
+        const queryParam: QueryParamObject = {
           filter: filterStrings,
           scope: scopeQueryParam
-        }
-      }))
+        };
+
+        return {
+          navType: NavigationParamsType.InApp,
+          path: '/explorer',
+          queryParams: !isNil(timeRange)
+            ? { ...queryParam, ...this.timeRangeService.toQueryParams(timeRange) }
+            : queryParam
```

Comment:
You don't need another block to do it. You can do it like this
```ts
    return forkJoinSafeEmpty(filterStrings$).pipe(
      map(filterStrings => ({
        navType: NavigationParamsType.InApp,
        path: '/explorer',
        queryParams: {
          filter: filterStrings,
          scope: scopeQueryParam,
          ...(!isNil(timeRange) ? this.timeRangeService.toQueryParams(timeRange) : {})
        }
      }))
```

Reply:
oh yeah, updated